src/main/java/duck/task/TaskList.java

```diff
@@ -0,0 +1,77 @@
+package duck.task;
+
+import java.util.ArrayList;
+
+public class TaskList {
+    private ArrayList<Task> taskList = new ArrayList<Task>();
+
+    /**
+     * initialize  the TaskLisk object
+     * @param data use data to form task list
+     */
+    public TaskList(String[] data) {
+        for (int i = 0; data[i] != null; i++) {
+            String datas = data[i];
+            String[] dataSplit = datas.split("\\|");
+            switch (dataSplit[0].trim()) {
+            case "T":
+                taskList.add(new Todo(dataSplit[2].trim()));
+                break;
+            case "D":
+                taskList.add(new Deadline(dataSplit[2].trim(), dataSplit[3].trim()));
```

Comment:
Perhaps for greater readability for future collaborations or debugging, the dataSplit could be assigned to multiple variables with a more intuitive name? E.g. taskDescription, taskDeadline